wgpu-core/src/lib.rs

```diff
@@ -123,7 +122,10 @@ pub type SubmissionIndex = hal::FenceValue;
 type Index = u32;
 type Epoch = u32;
 
-pub type RawString = *const c_char;
+#[cfg(feature = "std")]
+pub type RawString = *const std::os::raw::c_char;
+#[cfg(not(feature = "std"))]
+pub type RawString = *const u8; // TODO: changing this type alias is a bad plan
```

Comment:
This needs fixing, but whatever we do here (short of introducing cfgs to specify which of `u8` or `i8` to use here) will change the function signatures.

Presumably in actual FFI use of `bundle_ffi`, whether this is i8 or u8 doesn’t matter in practice, but it matters when Rust calls Rust (i.e. for the `wgpu` crate which is passing in a `CStr::as_ptr()` here). So, the obvious solution to me is to have separate interfaces for FFI and `wgpu`, but I don’t know if that is best.

Reply:
Is there a reason we can't use [`core::ffi::c_char`](https://doc.rust-lang.org/stable/core/ffi/type.c_char.html) unconditionally here? 🤔

Reply:
🤦 I saw `::os::` and figured “well obviously `core` hasn’t got any OS functionality so there is no replacement”. But `char` signedness is really part of the C ABI, isn’t it. Done.